File: src/code/js/_001.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    var map = {};
    for (var i = 0; i < nums.length; i++) {
        var n = target - nums[i];
        if (map[n] !== undefined) {
            return [map[n], i];
        }
        map[nums[i]] = i;
    }
};

File: src/code/java/_001.java
package code.java;

import java.util.Map;
import java.util.HashMap;

public class _001 {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++) {
            int n = target - nums[i];
            if (map.containsKey(n)) {
                return new int[] { map.get(n), i };
            }
            map.put(nums[i], i);
        }
        throw new RuntimeException("No solution");
    }
}
